### PR TITLE
Feat/angie deena/vault

### PR DESCRIPTION
**Pull request by Angie-n**
Vault page with the ability to upload & delete files with search & filter functionality 
Screenshot 2024-06-06 at 11 43 04 PM

### CODE REVIEW

File: source/project/__tests__/fileMgr.test.js
Comment (mapersiani) [inline source/project/__tests__/fileMgr.test.js:1]: Tests look good, glad you got the last on to work

**Review by mapersiani — APPROVED** [review]
Vault is fullt functional and the code is commented + passes all checks/tests. We might want to relocate all the images delete icon to the res folder (should relocate edit and delete for mainview to res folder too).

**Review by arnavkamdar — APPROVED** [review]
Changes look great to me, great job on the dev documentation! I agree with Matteo that some of our files could be better organized by moving them to the res folder, maybe do that before merging.

**Comment by mapersiani** [comment]
> Changes look great to me, great job on the dev documentation! I agree with Matteo that some of our files could be better organized by moving them to the res folder, maybe do that before merging.

Fixed

**Pull request merged**